mlflow/recipes/utils/wrapped_recipe_model.py

```diff
@@ -0,0 +1,34 @@
+import mlflow
+from mlflow.pyfunc import PythonModel
+import pandas as pd
+import numpy as np
+
+
+class WrappedRecipeModel(PythonModel):
+    def __init__(self, predict_scores_for_all_classes=True, prefix="predicted_"):
+        super().__init__()
+        self.predict_scores_for_all_classes = predict_scores_for_all_classes
+        self.prefix = prefix
+
+    def load_context(self, context):
+        self._classifier = mlflow.sklearn.load_model(context.artifacts["model_path"])
+
+    def predict(self, context, model_input):
+        predicted_label = self._classifier.predict(model_input)
+        if not hasattr(self._classifier, "classes_"):
+            return predicted_label
+
+        classes = self._classifier.classes_
+        score_cols = [f"{self.prefix}score_" + str(c) for c in classes]
+        output = {}
+        try:
+            probabilities = self._classifier.predict_proba(model_input)
+            if self.predict_scores_for_all_classes:
+                output = pd.DataFrame(columns=score_cols, data=probabilities)
+            output[f"{self.prefix}score"] = np.max(probabilities, axis=1)
+            output[f"{self.prefix}label"] = predicted_label
+
+            return output
+        except Exception:
+            # swallow that some models don't have predict_proba.
```

Comment:
Shouldn't we check to see if `hasattr(self._classifier, "predict_proba")`? That sounds better than try / catching all Exceptions